daml-foundations/daml-ghc/daml-stdlib-src/DA/TextMap.daml

```diff
@@ -74,15 +73,14 @@ insert = primitive @"BEMapInsert"
 union : TextMap a -> TextMap a -> TextMap a
 union m1 m2 = foldl (\a b -> (uncurry insert) b a) m2 (toList m1)
 
-instance (Show a) => Show (TextMap a) where
-  show m = "TextMap(" <> show (toList m) <> ")"
+instance Show a => Show (TextMap a) where
+  show m = "TextMap " <> show (toList m)
 
-
-instance (Eq a) => Eq (TextMap a) where
+instance Eq a => Eq (TextMap a) where
   x == y = toList x == toList y
 
-instance (Ord a) => Ord (TextMap a) where
-  compare = \m1 m2 -> compare (toList m1) (toList m2)
+instance Ord a => Ord (TextMap a) where
+  x `compare` y = toList x `compare` toList y
```

Comment:
We should add `comparing` to the stdlib.

Reply:
The question is where to put it. `DA.Ord`? Or do we want `on`? In `DA.Prelude` or `DA.Function`?

Reply:
`DA.Ord` sounds sensible.

Reply:
I've not found `comparing` to be that valuable in Haskell - I actually removed the HLint hints pointing people at it.